app/src/main/java/com/example/drawingApp/utils/DialogUtility.kt

```diff
@@ -30,6 +29,45 @@ object DialogUtility {
         val onDelete: (Int) -> Unit
     )
 
+    fun tabSheetDialog(view: LinearLayout, context: Context) {
+        val behavior = BottomSheetBehavior.from(view)
+        val colorPicker = view.findViewById<LinearLayout>(R.id.colorPickerLayout)
+        val editLayers = view.findViewById<LinearLayout>(R.id.editLayerLayout)
+        val hideTabs: () -> Unit = {
+            colorPicker.isVisible = false
+            editLayers.isVisible = false
+        }
+        val placeHolderFunc: () -> Unit = {
+            behavior.state = BottomSheetBehavior.STATE_EXPANDED
+        }
+        view.findViewById<ImageView>(R.id.colorGradientBtn).setOnClickListener {
+            hideTabs()
+            colorPicker.isVisible = true
+            placeHolderFunc()
+        }
+        view.findViewById<ImageView>(R.id.brushSettingsBtn).setOnClickListener {
+            hideTabs()
+            placeHolderFunc()
+        }
+        view.findViewById<ImageView>(R.id.moveImageBtn).setOnClickListener {
+            hideTabs()
+            placeHolderFunc()
+        }
+        view.findViewById<ImageView>(R.id.resizeBtn).setOnClickListener {
+            hideTabs()
+            placeHolderFunc()
+        }
+        view.findViewById<ImageView>(R.id.filtersBtn).setOnClickListener {
+            hideTabs()
+            placeHolderFunc()
+        }
+        view.findViewById<ImageView>(R.id.layersBtn).setOnClickListener {
```

Comment:
You are duplicating lots of code here. There are more elegant ways to reduce the duplication. Please try to slim it down